RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizerForBricked.cc

```diff
@@ -0,0 +1,229 @@
+          return cl1.minPixelRow() < cl2.minPixelRow();
+        });
+      }
+    }
+  }
+  // sort by row (x)   maybe sorting the seed would suffice....
+  std::sort_heap(output.begin(), output.end(), [](SiPixelCluster const& cl1, SiPixelCluster const& cl2) {
+    return cl1.minPixelRow() < cl2.minPixelRow();
+  });
+
+  // Erase the seeds.
+  theSeeds.clear();
+
+  //  Need to clean unused pixels from the buffer array.
+  clear_buffer(begin, end);
+
+  theFakePixels.clear();
+}
+
+//----------------------------------------------------------------------------
+//!  \brief The actual clustering algorithm: group the neighboring pixels around the seed.
+//----------------------------------------------------------------------------
+SiPixelCluster PixelThresholdClusterizerForBricked::make_cluster_bricked(
+    const SiPixelCluster::PixelPos& pix, edmNew::DetSetVector<SiPixelCluster>::FastFiller& output, bool isbarrel) {
+  //First we acquire the seeds for the clusters
+  int seed_adc;
+  stack<SiPixelCluster::PixelPos, vector<SiPixelCluster::PixelPos> > dead_pixel_stack;
+
+  //The individual modules have been loaded into a buffer.
+  //After each pixel has been considered by the clusterizer, we set the adc count to 1
+  //to mark that we have already considered it.
+  //The only difference between dead/noisy pixels and standard ones is that for dead/noisy pixels,
+  //We consider the charge of the pixel to always be zero.
+
+  seed_adc = theBuffer(pix.row(), pix.col());
+  theBuffer.set_adc(pix, 1);
+
+  AccretionCluster acluster;
+  acluster.add(pix, seed_adc);
+
+  //Here we search all pixels adjacent to all pixels in the cluster.
+  bool dead_flag = false;
+  while (!acluster.empty()) {
+    //This is the standard algorithm to find and add a pixel
+    auto curInd = acluster.top();
+    acluster.pop();
+
+    for (auto r = std::max(0, int(acluster.x[curInd]) - 1); r < std::min(int(acluster.x[curInd]) + 2, theBuffer.rows());
+         ++r) {
+      int LowerAccLimity = 0;
+      int UpperAccLimity = 0;
+
+      if (r % 2 == int(acluster.x[curInd]) % 2) {
+        LowerAccLimity = std::max(0, int(acluster.y[curInd]) - 1);
+        UpperAccLimity = std::min(int(acluster.y[curInd]) + 2, theBuffer.columns());
+      }
+
+      else {
+        int parity_curr = int(acluster.x[curInd]) % 2;
+        int parity_hit = r % 2;
+
+        LowerAccLimity = std::max(0, int(acluster.y[curInd]) - parity_hit);
+        UpperAccLimity = std::min(int(acluster.y[curInd]) + parity_curr + 1, theBuffer.columns());
+      }
+
+      /*
+	for (auto c = std::max(0, int(acluster.y[curInd]) - 1);
+	c < std::min(int(acluster.y[curInd]) + 2, theBuffer.columns());
+	++c)
+      */
+      for (auto c = LowerAccLimity; c < UpperAccLimity; ++c) {
+        if (theBuffer(r, c) >= thePixelThreshold) {
+          SiPixelCluster::PixelPos newpix(r, c);
+          if (!acluster.add(newpix, theBuffer(r, c)))
+            goto endClus;
```

Comment:
Could this `goto` be avoided?

Reply:
see https://github.com/cms-sw/cmssw/issues/34705.
A general solution needs to be found out (not particularly for this PR).